go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -274,13 +274,12 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 		return addErrors
 	}
 
-	if nsInfo.addressSet == nil {
-		// TODO(trozet): remove dependency on nsInfo object and just determine hash names to create Egress FW with
-		return fmt.Errorf("unable to add egress firewall, namespace: %s has no address set", egressFirewall.Namespace)
+	err = oc.addressSetFactory.EnsureAddressSet(egressFirewall.Namespace)
```

Comment:
@JacobTanenbaum shoudl have a code comment here that since EgressFirewall creates things (PortGroup? ACL?) that use AddressSets, EF must ensure the namespace address set exists independently of the Namespace object. And that all this should eventually go away if/when we refcount address sets or something like that.

Reply:
done